### PR TITLE
Added declarations for 'random-number

### PR DESCRIPTION
**Pull request by OpenByteDev**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/random-number/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+// TypeScript Version: 2.2
+
+type Generator = (min?: number|null, max?: number|null, integer?: boolean|null) => number;
+
+interface UnparsedOptions {
+    min?: number|null;
+    max?: number|null;
+    integer?: boolean|null;
```
Comment (DanielRosenwasser) [inline types/random-number/index.d.ts]: Is there a reason these are potentially `null`?

Comment (OpenByteDev) [reply types/random-number/index.d.ts]: If one of these properties is `null` or `undefined` the library sets the value to a default.
There is maybe some edge case where you pass user inputs as arguments, which are `null` when the default should be used:

```ts
import rn = require('random-number');

function getNumberInput(message: string): number|null { }
function getBooleanInput(message: string): boolean|null { }

const num = rn({
    min: getNumberInput('min'),
    max: getNumberInput('max'),
    integer: getBooleanInput('boolean')
});
```

My goal was to match the library, so I included it, If you think it shouldn't be this way I have no problem removing it.

Comment (DanielRosenwasser) [reply types/random-number/index.d.ts]: Yeah, it's definitely a sensitive balance. I think it probably catches more being more conservative and disallowing `null`. If users end up using it, they can always make it more permissive.

**Comment by DanielRosenwasser** [comment]
Thank you!

**Pull request merged**